web-examples/src/main/java/io/vertx/example/web/jdbc/Server.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.example.web.jdbc;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.example.util.Runner;
+import io.vertx.ext.jdbc.JDBCClient;
+import io.vertx.ext.sql.SQLConnection;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+
+/**
+ * @author <a href="mailto:pmlopes@gmail.com">Paulo Lopes</a>
+ */
+public class Server extends AbstractVerticle {
+
+  // Convenience method so you can run it in your IDE
+  public static void main(String[] args) {
+    Runner.runExample(Server.class);
+  }
+
+  private JDBCClient client;
+
+  @Override
+  public void start() {
+
+    // Create a JDBC client with a test database
+    client = JDBCClient.createShared(vertx, new JsonObject()
+      .put("url", "jdbc:hsqldb:mem:test?shutdown=true")
+      .put("driver_class", "org.hsqldb.jdbcDriver"));
+
+    setUpInitialData(ready -> {
+      Router router = Router.router(vertx);
+
+      router.route().handler(BodyHandler.create());
+      router.get("/products/:productID").handler(Server.this::handleGetProduct);
+      router.post("/products").handler(Server.this::handleAddProduct);
+      router.get("/products").handler(Server.this::handleListProducts);
+
+      vertx.createHttpServer().requestHandler(router::accept).listen(8080);
+    });
+  }
+
+  private void handleGetProduct(RoutingContext routingContext) {
+    String productID = routingContext.request().getParam("productID");
+    HttpServerResponse response = routingContext.response();
+    if (productID == null) {
+      sendError(400, response);
+    } else {
+      client.getConnection(res -> {
```

Comment:
Instead of duplicating client.getConnection() in each method it would be nice to have a single method which gets the connection and into which you pass the lambda which does the action.